prac_05/hex_colour.py

```diff
@@ -0,0 +1,26 @@
+"""Program to look up hexadecimal colour codes."""
+HEX_COLOURS = {"red1": "#ff0000", "purple": "#9020f0", "SpringGreen2": "#00ee76", "turquoise1": "#00f5ff", "yellow1": "#ffff00", "orchid1": "#ff83fa", "DeepSkyBlue1": "#00bfff", "chocolate1": "#ff7f24", "black": "#000000", "white": "#ffffff"}
+
+
+def main ():
+    """Program to return hexadecimal colour codes when colour name is entered."""
+    colour = str(input("Enter colour name: "))
+    while not is_colour_name_valid(colour):
```

Comment:
You are able to heavily simplify this code by using a nested if statement within a while loop.  
Example:
colour = input
while colour != ""
if statement

Reply:
Thanks Mitchell. I've updated the code with your recommendations. I knew there had to be a better way just couldn't work it out.